### ISSUE TITLE
EasyEngine breaks default sitemap locations

### ISSUE THREAD

**Issue by sybrew** [sybrew/the-seo-framework#107]
In https://github.com/EasyEngine/easyengine/issues/601, there's described that default rewrite rules point to `sitemap_index.xml` rather than `sitemap.xml`. 

To fix this, we need to do the following:
1. Output the sitemap regardless on `sitemap_index.xml`.
2. Maybe add support for `robots.txt` sitemap location. Although the 301 automatically takes care of this.
3. Let Monitor check for this. This unconventional placement is actually an issue.

This is an issue with EasyEngine, but it's very annoyingly a broad issue they're unlikely to fix.

**Comment by RavanH** [comment]
Indeed, the redirect to sitemap_index.xml breaks other XML Sitemap plugins like XML Sitemap Feed or the XML Sitemap module in Jetpack.

Robots.txt is supported as far as I know...

**Comment by kevinlisota** [comment]
For any plugin that is dynamically generating sitemap.xml files or in cases where wordpress is dynamically generating robots.txt, there should actually be no rewrite rules present in nginx at all. In fact, Yoast's own KB article gives the incorrect advice for nginx.

Your comment on this was correct. try_files takes care of this.
```
location / {
        try_files $uri $uri/ /index.php?$args;
}
```
The Yoast rewrite rules are a lazy way around this, and rewrites are actually not the optimal way in nginx and are actually slower than location blocks and try_files. There should be location blocks with a return 301, if needed, though in the case of passing files to WordPress, it should just be done with the WP Rewrite API with no rewriting in nginx at all.

Anyway, I wouldn't change your sitemap locations to accommodate this. The fix is folks tweaking the nginx.conf files on their servers when needed, and for Yoast to learn a little bit more about nginx configuration :)

**Comment by sybrew** [comment]
> I wouldn't change your sitemap locations to accommodate this.

Neither would I in a perfect world. 
Unfortunately, I can't wait for the government to fix the hole in the road whilst they assume most won't be bothered; I need to upgrade my bike to a car instead for if I want to keep up with my companions.

**Comment by sybrew** [comment]
Fixed in EE 4.0 😃

**Comment by sybrew** [comment]
Other known offending parties, which force users away, are:
- EasyWP
- SpinupWP 
- Namecheap

So, in https://github.com/sybrew/the-seo-framework/commit/7e2e3da84eff62285acc5e6fa9541938e065b672#diff-818889a7b6451244983ffec8b63dfd0e we added support for Yoast SEO's cancerous rules, anyway.
